improver/cli/apply_emos_coefficients.py

```diff
@@ -153,6 +153,10 @@ def process(cube: cli.inputcube,
 
     current_forecast = cube
 
+    if current_forecast.name() == 'emos_coefficients':
+        msg = "The current forecast cube has the name 'emos_coefficients'"
+        raise ValueError(msg)
+
```

Comment:
This solves the current problem but I don't think it fully covers the range of possible / probable errors.

There are up to three input files; the first being compulsory, the next two being optional and positional.  So there is another thing that can go wrong: the land-sea mask can appear in the wrong place, and be labelled as either a forecast cube or a coefficients cube, depending on which input arguments exist.

Can you extend this error handling to cover the scope of possible input errors given either or both of the forecast and coefficient files being missing, please?

Reply:
I think there are two ways of fixing this. I've tried what I think is the simplest and pushed it up for you to look at.
I think the alternative is not something I can do now. I think this will involve:
* Passing in a cubelist rather than a cube into the CLI
* Filtering out the land sea mask and coefficients cubes if they exist.
* Check that there is at least one file that is not a land sea mask or a coefficients cube, but we don't know the name of this cube.
* If no coefficients cube return the original forecast file, otherwise carry on.

I don't think we have an example of doing anything particularly similar to this in the current CLIs.